modules/images/webp-uploads/load.php

```diff
@@ -35,5 +35,186 @@ function webp_uploads_filter_image_editor_output_format( $output_format, $filena
+		}
+
+		$current_size = array();
+		if ( array_key_exists( $size, $image_sizes ) && is_array( $image_sizes[ $size ] ) ) {
+			$current_size = $image_sizes[ $size ];
+		}
+
+		$sources = array();
+		if ( array_key_exists( 'sources', $current_size ) && is_array( $current_size['sources'] ) ) {
+			$sources = $current_size['sources'];
+		}
+
+		// Try to find the mime type of the image size.
+		if ( array_key_exists( 'mime-type', $current_size ) ) {
+			$current_mime = $current_size['mime-type'];
+		} elseif ( array_key_exists( 'file', $current_size ) ) {
+			$current_mime = wp_check_filetype( $current_size['file'] )['type'];
+		} else {
+			$current_mime = '';
+		}
+
+		// The mime for this file couldn't be determined.
+		if ( empty( $current_mime ) ) {
+			continue;
+		}
+
+		// Make sure the current mime is consider a valid mime type.
+		if ( ! array_key_exists( $current_mime, $valid_mime_types ) ) {
+			continue;
+		}
+
+		$sources[ $current_mime ] = array(
+			'file' => array_key_exists( 'file', $current_size ) ? $current_size['file'] : '',
+			// TOOD: Add filesize from the original version of this image.
+		);
+
+		$formats = array_diff_assoc( $valid_mime_types, array( $current_mime => $valid_mime_types[ $current_mime ] ) );
+
+		foreach ( $formats as $mime => $extension ) {
+			// Editor needs to be recreated every time as there is not flush() or clear() function that can be used after we created an image.
+			$editor = wp_get_image_editor( $file );
+
+			if ( is_wp_error( $editor ) ) {
+				continue;
+			}
+
+			$editor->resize( (int) $properties['width'], (int) $properties['height'], $properties['crop'] );
+			$filename = $editor->generate_filename( null, null, $extension );
+			$image    = $editor->save( $filename, $mime );
+
+			if ( is_wp_error( $image ) ) {
+				continue;
+			}
+
+			// TODO: Store the file size of the created image.
+			// $image['filesize'] = filesize( $image['path'] );
+			// Remove duplicated properties from the size image.
+			unset( $image['path'], $image['height'], $image['width'], $image['mime-type'] );
+
+			$sources[ $mime ] = $image;
+		}
+
+		$current_size['sources'] = $sources;
+		$sizes[ $size ]          = $current_size;
+	}
+
+	$metadata['sizes'] = $sizes;
+
+	return $metadata;
+}
+
+/**
+ * List all the available image sizes as well with the properties for each, the properties included
+ * are: width, height and crop. The logic behind this function tries to use the defined sizes by
+ * WordPress if the values is not found on the options it would fallback to the user defined
+ * sizes. Each property uses the default for each property `null` for height and width and `false`
+ * for crop the same way as `make_subsize` the editor image.
+ *
+ * @since n.e.x.t
+ *
+ * @see   WP_Image_Editor_Imagick::make_subsize()
+ * @see   WP_Image_Editor_GD::make_subsize()
+ * @return array An array with the details of all available image sizes: width, height and crop.
+ */
+function webp_uploads_get_image_sizes() {
```

Comment:
I don't think this function is needed. WordPress core already has [`wp_get_registered_image_subsizes()`](https://developer.wordpress.org/reference/functions/wp_get_registered_image_subsizes/) which should allow us to get exactly this data.

Reply:
Great I wasn't aware of that one.